// tailwind-mobile/react/cjs/components/Card.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _react = _interopRequireWildcard(require("react"));

var _cls = require("../shared/cls.js");

var _useDarkClasses = require("../shared/use-dark-classes.js");

var _useThemeClasses = require("../shared/use-theme-classes.js");

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function (nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

const Card = /*#__PURE__*/(0, _react.forwardRef)((props, ref) => {
  const {
    component = 'div',
    className,
    margin = 'm-4',
    colors: colorsProp,
    header,
    footer,
    ios,
    material,
    // Style
    outline,
    // Children
    children,
    // Rest
    ...rest
  } = props;
  const elRef = (0, _react.useRef)(null);
  (0, _react.useImperativeHandle)(ref, () => ({
    el: elRef.current
  }));
  const Component = component;
  const attrs = { ...rest
  };
  const themeClasses = (0, _useThemeClasses.useThemeClasses)({
    ios,
    material
  });
  const dark = (0, _useDarkClasses.useDarkClasses)();
  const colors = {
    bg: (0, _cls.cls)(`bg-block-strong-light`, dark('dark:bg-block-strong-dark')),
    footerText: (0, _cls.cls)(`text-black`, dark('dark:text-white')),
    ...colorsProp
  };
  const style = outline ? 'outline' : 'shadow';
  const c = themeClasses({
    base: {
      common: (0, _cls.cls)(margin, colors.bg, `rounded overflow-hidden`),
      shadow: 'shadow',
      outline: (0, _cls.cls)(`border border-black border-opacity-10`, dark('dark:border-white dark:border-opacity-10'))
    },
    header: (0, _cls.cls)(`border-b border-black border-opacity-10 px-4 py-2`, dark('dark:border-white dark:border-opacity-10')),
    content: 'p-4 text-sm',
    footer: (0, _cls.cls)(`border-t border-black border-opacity-10 text-sm text-opacity-55`, colors.footerText, dark('dark:border-white  dark:border-opacity-10 dark:text-opacity-55 px-4 py-2'))
  }, className);
  return /*#__PURE__*/_react.default.createElement(Component, _extends({
    ref: elRef,
    className: c.base[style]
  }, attrs), header && /*#__PURE__*/_react.default.createElement("div", {
    className: c.header
  }, header), /*#__PURE__*/_react.default.createElement("div", {
    className: c.content
  }, children), footer && /*#__PURE__*/_react.default.createElement("div", {
    className: c.footer
  }, footer));
});
Card.displayName = 'Card';
var _default = Card;
exports.default = _default;
