// tailwind-mobile/react/cjs/components/Navbar.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _react = _interopRequireWildcard(require("react"));

var _cls = require("../shared/cls.js");

var _positionClass = require("../shared/position-class.js");

var _useThemeClasses = require("../shared/use-theme-classes.js");

var _useDarkClasses = require("../shared/use-dark-classes.js");

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function (nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

const Navbar = /*#__PURE__*/(0, _react.forwardRef)((props, ref) => {
  const {
    component = 'div',
    className,
    bgClassName = '',
    innerClassName = '',
    leftClassName = '',
    titleClassName = '',
    subtitleClassName = '',
    rightClassName = '',
    subnavbarClassName = '',
    colors: colorsProp,
    translucent = true,
    hairlines = true,
    fontSizeIos = 'text-navbar-ios',
    fontSizeMaterial = 'text-navbar-material',
    left,
    title,
    subtitle,
    right,
    subnavbar,
    ios,
    material,
    children,
    ...rest
  } = props;
  const elRef = (0, _react.useRef)(null);
  (0, _react.useImperativeHandle)(ref, () => ({
    el: elRef.current
  }));
  const Component = component;
  const themeClasses = (0, _useThemeClasses.useThemeClasses)({
    ios,
    material
  });
  const dark = (0, _useDarkClasses.useDarkClasses)();
  const attrs = { ...rest
  };
  const colors = {
    bgIos: (0, _cls.cls)('bg-bars-ios-light', dark('dark:bg-bars-ios-dark')),
    bgMaterial: (0, _cls.cls)('bg-bars-material-light', dark('dark:bg-bars-material-dark')),
    title: (0, _cls.cls)('text-black', dark('dark:text-white')),
    ...colorsProp
  };
  const c = themeClasses({
    base: {
      common: (0, _cls.cls)(`w-full z-20 top-0 pt-safe`, (0, _positionClass.positionClass)('sticky', className)),
      ios: fontSizeIos,
      material: fontSizeMaterial
    },
    bg: {
      common: (0, _cls.cls)('absolute w-full h-full left-0 top-0', bgClassName),
      ios: (0, _cls.cls)(colors.bgIos, hairlines && 'hairline-b', translucent && 'translucent'),
      material: `shadow-md ${colors.bgMaterial}`
    },
    subnavbar: {
      common: (0, _cls.cls)('relative flex items-center', subnavbarClassName),
      ios: 'h-11 pl-2-safe pr-2-safe',
      material: 'h-12 pl-4-safe pr-4-safe'
    },
    inner: {
      common: (0, _cls.cls)('flex relative items-center w-full overflow-hidden', innerClassName),
      ios: (0, _cls.cls)('pl-2-safe pr-2-safe h-11', !left && right ? 'justify-end' : 'justify-between'),
      material: 'justify-start h-14 lg:h-16 pl-safe pr-safe'
    },
    left: {
      common: (0, _cls.cls)('flex justify-center items-center h-full', leftClassName),
      ios: 'mr-2 transform transform-gpu',
      material: 'mx-1'
    },
    title: {
      common: (0, _cls.cls)(`${colors.title} whitespace-nowrap leading-tight`, titleClassName),
      ios: 'font-semibold absolute top-1/2 left-1/2 transform-gpu -translate-x-1/2 -translate-y-1/2 text-center',
      material: 'font-medium mx-4 text-left'
    },
    subtitle: {
      common: (0, _cls.cls)('font-normal leading-none', subtitleClassName),
      ios: 'text-2xs opacity-55',
      material: 'text-sm opacity-85'
    },
    right: {
      common: (0, _cls.cls)('flex justify-center items-center h-full', rightClassName),
      ios: 'ml-2 transform transform-gpu',
      material: 'ml-auto mr-1'
    }
  }, className);
  return /*#__PURE__*/_react.default.createElement(Component, _extends({
    ref: elRef,
    className: c.base
  }, attrs), /*#__PURE__*/_react.default.createElement("div", {
    className: c.bg
  }), /*#__PURE__*/_react.default.createElement("div", {
    className: c.inner
  }, left && /*#__PURE__*/_react.default.createElement("div", {
    className: c.left
  }, left), (title || subtitle) && /*#__PURE__*/_react.default.createElement("div", {
    className: c.title
  }, title, subtitle && /*#__PURE__*/_react.default.createElement("div", {
    className: c.subtitle
  }, subtitle)), right && /*#__PURE__*/_react.default.createElement("div", {
    className: c.right
  }, right), children), subnavbar && /*#__PURE__*/_react.default.createElement("div", {
    className: c.subnavbar
  }, subnavbar));
});
Navbar.displayName = 'Navbar';
var _default = Navbar;
exports.default = _default;
