// tailwind-mobile/react/cjs/components/Toolbar.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _react = _interopRequireWildcard(require("react"));

var _cls = require("../shared/cls.js");

var _positionClass = require("../shared/position-class.js");

var _useThemeClasses = require("../shared/use-theme-classes.js");

var _useDarkClasses = require("../shared/use-dark-classes.js");

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function (nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

const Toolbar = /*#__PURE__*/(0, _react.forwardRef)((props, ref) => {
  const {
    component = 'div',
    className,
    colors: colorsProp,
    translucent = true,
    bgClassName = '',
    innerClassName = '',
    hairlines = true,
    tabbar,
    tabbarLabels,
    top,
    ios,
    material,
    children,
    ...rest
  } = props;
  const elRef = (0, _react.useRef)(null);
  (0, _react.useImperativeHandle)(ref, () => ({
    el: elRef.current
  }));
  const Component = component;
  const themeClasses = (0, _useThemeClasses.useThemeClasses)({
    ios,
    material
  });
  const dark = (0, _useDarkClasses.useDarkClasses)();
  const attrs = { ...rest
  };
  const colors = {
    bgIos: (0, _cls.cls)('bg-bars-ios-light', dark('dark:bg-bars-ios-dark')),
    bgMaterial: (0, _cls.cls)('bg-bars-material-light', dark('dark:bg-bars-material-dark')),
    ...colorsProp
  };
  const c = themeClasses({
    base: {
      common: (0, _cls.cls)(`w-full z-20`, (0, _positionClass.positionClass)('relative', className), !top && 'pb-safe')
    },
    bg: {
      common: (0, _cls.cls)('absolute w-full h-full left-0 top-0', bgClassName),
      ios: (0, _cls.cls)(hairlines && (top ? 'hairline-b' : 'hairline-t'), colors.bgIos, translucent && 'translucent'),
      material: (0, _cls.cls)(`shadow-md ${colors.bgMaterial}`, !top && 'transform rotate-180')
    },
    inner: {
      common: (0, _cls.cls)(`flex relative justify-between items-center w-full overflow-hidden`, innerClassName),
      ios: 'pl-2-safe pr-2-safe',
      material: !tabbar ? 'pl-2-safe pr-2-safe' : '',
      toolbar: {
        ios: 'h-11',
        material: 'h-12'
      },
      tabbarLabels: {
        ios: 'h-12.5',
        material: 'h-14'
      }
    }
  }, className);
  return /*#__PURE__*/_react.default.createElement(Component, _extends({
    ref: elRef,
    className: c.base
  }, attrs), /*#__PURE__*/_react.default.createElement("div", {
    className: c.bg
  }), /*#__PURE__*/_react.default.createElement("div", {
    className: c.inner[tabbar && tabbarLabels ? 'tabbarLabels' : 'toolbar']
  }, children));
});
Toolbar.displayName = 'Toolbar';
var _default = Toolbar;
exports.default = _default;
