// tailwind-mobile/react/esm/components/BlockTitle.js
function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

import React, { useRef, forwardRef, useImperativeHandle } from 'react';
import { useThemeClasses } from '../shared/use-theme-classes.js';
const BlockTitle = /*#__PURE__*/forwardRef((props, ref) => {
  const {
    component = 'div',
    className,
    withBlock = true,
    ios,
    material,
    // Children
    children,
    // Rest
    ...rest
  } = props;
  const elRef = useRef(null);
  useImperativeHandle(ref, () => ({
    el: elRef.current
  }));
  const Component = component;
  const attrs = { ...rest
  };
  const themeClasses = useThemeClasses({
    ios,
    material
  });
  const c = themeClasses({
    base: {
      common: `pl-4-safe pr-4-safe mt-8 flex justify-between items-center ${withBlock ? '-mb-6' : 'mb-2'}`,
      ios: `font-semibold`,
      material: `font-medium`
    }
  }, className);
  return /*#__PURE__*/React.createElement(Component, _extends({
    ref: elRef,
    className: c.base
  }, attrs), children);
});
BlockTitle.displayName = 'BlockTitle';
export default BlockTitle;
