// tailwind-mobile/react/esm/components/ListItem.js
function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

import React, { useRef, forwardRef, useImperativeHandle } from 'react';
import { cls } from '../shared/cls.js';
import { useTheme } from '../shared/use-theme.js';
import { useThemeClasses } from '../shared/use-theme-classes.js';
import { useTouchRipple } from '../shared/use-touch-ripple.js';
import ChevronIcon from './icons/ChevronIcon.js';
import { useDarkClasses } from '../shared/use-dark-classes.js';
const ListItem = /*#__PURE__*/forwardRef((props, ref) => {
  const {
    component = 'li',
    colors: colorsProp,
    className,
    mediaClassName = '',
    innerClassName = '',
    innerChildren,
    contentClassName = '',
    contentChildren,
    titleWrapClassName = '',
    titleFontSizeIos = 'text-list-title-ios',
    titleFontSizeMaterial = 'text-list-title-material',
    // Content props
    title,
    subtitle,
    text,
    after,
    media,
    header,
    footer,
    menuListItem,
    menuListItemActive,
    hairlines = true,
    // Enable divider
    divider,
    groupTitle,
    // Title
    strongTitle = 'auto',
    // Label props
    label,
    // Link props
    chevron = true,
    chevronIcon,
    href,
    target,
    link,
    linkComponent = 'a',
    linkProps = {},
    ios,
    material,
    touchRipple = true,
    // Children
    children,
    // Rest
    ...rest
  } = props;
  const rippleElRef = useRef(null);
  const elRef = useRef(null);
  useImperativeHandle(ref, () => ({
    el: elRef.current
  }));
  const Component = component;
  const attrs = { ...rest
  };
  const theme = useTheme({
    ios,
    material
  });
  const themeClasses = useThemeClasses({
    ios,
    material
  });
  const dark = useDarkClasses();
  const colors = {
    text: cls(`text-black`, dark('dark:text-white')),
    menuListItemText: cls(`text-primary`, dark('dark:text-white')),
    menuListItemActiveBg: 'bg-primary dark:bg-primary',
    ...colorsProp
  };
  const isMenuListItemActive = menuListItem && menuListItemActive;
  const textColor = isMenuListItemActive ? colors.menuListItemText : colors.text;
  const isLink = !!href || href === '' || menuListItem || link;
  const isLabel = !!label;
  const needsTouchRipple = theme === 'material' && (isLabel || isLink) && touchRipple;
  useTouchRipple(rippleElRef, needsTouchRipple);
  const hrefComputed = href === true || href === false || typeof href === 'undefined' ? undefined : href || '';
  const ItemContentComponent = isLink ? linkComponent : isLabel ? 'label' : 'div';
  const linkPropsComputed = isLink ? {
    href: hrefComputed,
    target,
    ...linkProps
  } : {};
  const autoStrongTitle = strongTitle === 'auto' && title && (subtitle || text);
  const c = themeClasses({
    base: menuListItem ? `${textColor} py-1` : '',
    itemContent: {
      common: cls(menuListItem ? 'pl-2 ml-2-safe mr-2-safe rounded-lg' : 'pl-4-safe', `flex items-center ${contentClassName}`),
      link: cls('duration-300 active:duration-0 cursor-pointer select-none', hairlines && 'active:hairline-transparent', needsTouchRipple && cls(`relative overflow-hidden`, dark('dark:touch-ripple-white z-10')), isMenuListItemActive ? cls(colors.menuListItemActiveBg, 'bg-opacity-15') : cls(`active:bg-black active:bg-opacity-10`, dark('dark:active:bg-white dark:active:bg-opacity-10')))
    },
    media: {
      common: `mr-4 flex-shrink-0 ${mediaClassName}`,
      ios: 'py-2',
      material: 'py-3 min-w-10'
    },
    inner: {
      common: cls('pr-4-safe w-full relative', !menuListItem && hairlines && 'hairline-b', innerClassName),
      ios: 'py-2.5',
      material: 'py-3'
    },
    titleWrap: {
      common: cls(`flex justify-between items-center ${titleWrapClassName}`),
      ios: cls(!menuListItem && titleFontSizeIos),
      material: cls(!menuListItem && titleFontSizeMaterial)
    },
    title: {
      common: `flex-shrink`,
      menuListItem: cls('text-sm', strongTitle === true || autoStrongTitle ? 'font-semibold' : 'font-medium'),
      strong: {
        common: '',
        ios: 'font-semibold',
        material: 'font-medium'
      }
    },
    after: cls(textColor, `text-opacity-55 flex-shrink-0 ml-auto pl-1 flex items-center space-x-1`, dark('dark:text-opacity-55')),
    chevron: 'opacity-20 flex-shrink-0 ml-3',
    subtitle: 'text-sm',
    text: cls(textColor, `text-sm text-opacity-55 line-clamp-2`, dark('dark:text-opacity-55')),
    header: 'text-xs mb-0.5',
    footer: cls(textColor, `text-xs text-opacity-55 mt-0.5`, dark('dark:text-opacity-55')),
    divider: {
      common: cls(`bg-list-divider-light text-black text-opacity-55 pl-4-safe pr-4-safe py-1 flex items-center z-20`, divider ? 'relative' : 'sticky top-0', dark(`dark:bg-list-divider-dark dark:text-white dark:text-opacity-55`)),
      ios: `h-8${hairlines ? ' hairline-t' : ''} -mt-px text-list-title-ios`,
      material: 'h-12 text-list-title-material'
    }
  }, className);
  const itemContentClasses = isLink || isLabel ? c.itemContent.link : c.itemContent.default;
  const titleClasses = menuListItem ? c.title.menuListItem : strongTitle === true || autoStrongTitle ? c.title.strong : c.title.default;

  if (divider || groupTitle) {
    return /*#__PURE__*/React.createElement(Component, {
      className: cls(c.divider, className)
    }, title, children);
  }

  return /*#__PURE__*/React.createElement(Component, _extends({
    ref: elRef,
    className: c.base
  }, attrs), /*#__PURE__*/React.createElement(ItemContentComponent, _extends({
    ref: rippleElRef,
    className: itemContentClasses
  }, linkPropsComputed), media && /*#__PURE__*/React.createElement("div", {
    className: c.media
  }, media), /*#__PURE__*/React.createElement("div", {
    className: c.inner
  }, header && /*#__PURE__*/React.createElement("div", {
    className: c.header
  }, header), (title || after) && /*#__PURE__*/React.createElement("div", {
    className: c.titleWrap
  }, title && /*#__PURE__*/React.createElement("div", {
    className: titleClasses
  }, title), after && /*#__PURE__*/React.createElement("div", {
    className: c.after
  }, after), isLink && chevron && !menuListItem && (chevronIcon || /*#__PURE__*/React.createElement(ChevronIcon, {
    className: c.chevron
  }))), subtitle && /*#__PURE__*/React.createElement("div", {
    className: c.subtitle
  }, subtitle), text && /*#__PURE__*/React.createElement("div", {
    className: c.text
  }, text), footer && /*#__PURE__*/React.createElement("div", {
    className: c.footer
  }, footer), innerChildren), contentChildren), children);
});
ListItem.displayName = 'ListItem';
export default ListItem;
