// tailwind-mobile/vue/cjs/components/App.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _vue = require("vue");

var _cls = require("../shared/cls.js");

var _positionClass = require("../shared/position-class.js");

var _useAutoTheme = require("../shared/use-auto-theme.js");

var _TailwindMobileProvider = _interopRequireDefault(require("../shared/TailwindMobileProvider.js"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function render(_ctx, _cache) {
  const _component_tailwind_mobile_provider = (0, _vue.resolveComponent)("tailwind-mobile-provider");

  return (0, _vue.openBlock)(), (0, _vue.createBlock)((0, _vue.resolveDynamicComponent)(_ctx.component), {
    class: (0, _vue.normalizeClass)(_ctx.classes)
  }, {
    default: (0, _vue.withCtx)(() => [(0, _vue.createVNode)(_component_tailwind_mobile_provider, {
      theme: _ctx.currentTheme,
      dark: _ctx.dark,
      "touch-ripple": _ctx.touchRipple,
      "auto-theme-detection": false
    }, {
      default: (0, _vue.withCtx)(() => [(0, _vue.renderSlot)(_ctx.$slots, "default")]),
      _: 3
    }, 8, ["theme", "dark", "touch-ripple"])]),
    _: 3
  }, 8, ["class"]);
}

var _default = {
  name: 'twm-app',
  render,
  components: {
    TailwindMobileProvider: _TailwindMobileProvider.default
  },
  props: {
    component: {
      type: String,
      default: 'div'
    },
    theme: {
      type: String,
      default: 'material'
    },
    dark: {
      type: Boolean,
      default: true
    },
    touchRipple: {
      type: Boolean,
      default: true
    },
    safeAreas: {
      type: Boolean,
      default: true
    }
  },

  setup(props, ctx) {
    const currentTheme = (0, _useAutoTheme.useAutoTheme)(props);
    const classes = (0, _vue.computed)(() => (0, _cls.cls)(currentTheme.value === 'ios' && `twm-ios`, currentTheme.value === 'material' && 'twm-material', 'twm-app w-full h-full min-h-screen', props.safeAreas && 'safe-areas', (0, _positionClass.positionClass)('relative', ctx.attrs.class)));
    return {
      currentTheme,
      classes
    };
  }

};
exports.default = _default;
