// tailwind-mobile/vue/cjs/components/Checkbox.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _vue = require("vue");

var _cls = require("../shared/cls.js");

var _positionClass = require("../shared/position-class.js");

var _useDarkClasses = require("../shared/use-dark-classes.js");

var _useThemeClasses = require("../shared/use-theme-classes.js");

var _CheckboxIcon = _interopRequireDefault(require("./icons/CheckboxIcon.js"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

const _hoisted_1 = ["name", "value", "disabled", "readonly", "checked"];

function render(_ctx, _cache) {
  const _component_CheckboxIcon = (0, _vue.resolveComponent)("CheckboxIcon");

  return (0, _vue.openBlock)(), (0, _vue.createBlock)((0, _vue.resolveDynamicComponent)(_ctx.component), {
    class: (0, _vue.normalizeClass)(_ctx.c.base)
  }, {
    default: (0, _vue.withCtx)(() => [(0, _vue.createElementVNode)("input", {
      ref: "inputElRef",
      type: "checkbox",
      name: _ctx.name,
      value: _ctx.value,
      disabled: _ctx.disabled,
      readonly: _ctx.readonly,
      checked: _ctx.checked,
      class: (0, _vue.normalizeClass)(_ctx.c.input),
      onChange: _cache[0] || (_cache[0] = (...args) => _ctx.onChange && _ctx.onChange(...args))
    }, null, 42, _hoisted_1), (0, _vue.createElementVNode)("i", {
      class: (0, _vue.normalizeClass)(_ctx.c.iconWrap[_ctx.state])
    }, [_ctx.indeterminate ? ((0, _vue.openBlock)(), (0, _vue.createElementBlock)("span", {
      key: 0,
      class: (0, _vue.normalizeClass)(_ctx.c.indeterminateIcon)
    }, null, 2)) : ((0, _vue.openBlock)(), (0, _vue.createBlock)(_component_CheckboxIcon, {
      key: 1,
      ios: _ctx.ios,
      material: _ctx.material,
      class: (0, _vue.normalizeClass)(_ctx.c.icon[_ctx.state])
    }, null, 8, ["ios", "material", "class"]))], 2), (0, _vue.renderSlot)(_ctx.$slots, "default")]),
    _: 3
  }, 8, ["class"]);
}

var _default = {
  name: 'twm-checkbox',
  render,
  components: {
    CheckboxIcon: _CheckboxIcon.default
  },
  props: {
    component: {
      type: String,
      default: 'label'
    },
    colors: {
      type: Object
    },
    ios: {
      type: Boolean,
      default: undefined
    },
    material: {
      type: Boolean,
      default: undefined
    },
    checked: {
      type: Boolean,
      default: false
    },
    indeterminate: {
      type: Boolean,
      default: false
    },
    name: {
      type: String,
      default: undefined
    },
    value: {
      type: [Number, String],
      default: undefined
    },
    disabled: {
      type: Boolean,
      default: false
    },
    readonly: {
      type: Boolean,
      default: false
    }
  },
  emits: ['change'],

  setup(props, ctx) {
    const inputElRef = (0, _vue.ref)(null);
    const colors = (0, _vue.computed)(() => ({
      borderIos: (0, _cls.cls)(`border-black border-opacity-30`, (0, _useDarkClasses.useDarkClasses)('dark:border-white dark:border-opacity-30')),
      borderMaterial: (0, _cls.cls)(`border-black border-opacity-40`, (0, _useDarkClasses.useDarkClasses)('dark:border-white dark:border-opacity-40')),
      bgChecked: 'bg-primary',
      borderChecked: 'border-primary',
      ...(props.colors || {})
    }));
    const state = (0, _vue.computed)(() => props.checked || props.indeterminate ? 'checked' : 'notChecked');
    const c = (0, _useThemeClasses.useThemeClasses)(props, () => ({
      base: `cursor-pointer inline-block align-middle ${(0, _positionClass.positionClass)('relative', ctx.attrs.class)}`,
      iconWrap: {
        common: (0, _cls.cls)(`flex items-center justify-center text-white`, (0, _useDarkClasses.useDarkClasses)('dark:text-black')),
        ios: 'w-5.5 h-5.5 rounded-full border',
        material: 'w-4.5 h-4.5 rounded-sm border-2',
        notChecked: {
          ios: colors.value.borderIos,
          material: colors.value.borderMaterial
        },
        checked: `${colors.value.bgChecked} ${colors.value.borderChecked}`
      },
      icon: {
        notChecked: 'opacity-0',
        checked: 'opacity-100'
      },
      indeterminateIcon: {
        common: (0, _cls.cls)(`bg-white w-3/4`, (0, _useDarkClasses.useDarkClasses)('dark:bg-black')),
        ios: 'h-0.5 rounded-full',
        material: 'h-0.5'
      },
      input: 'hidden'
    }));
    (0, _vue.watch)(() => props.indeterminate, newValue => {
      inputElRef.value.indeterminate = !!newValue;
    });
    (0, _vue.onMounted)(() => {
      if (props.indeterminate) inputElRef.value.indeterminate = !!props.indeterminate;
    });

    const onChange = e => {
      ctx.emit('change', e);
    };

    return {
      inputElRef,
      c,
      state,
      onChange
    };
  }

};
exports.default = _default;
