// tailwind-mobile/vue/cjs/components/Link.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _vue = require("vue");

var _cls = require("../shared/cls.js");

var _positionClass = require("../shared/position-class.js");

var _useTheme = require("../shared/use-theme.js");

var _useThemeClasses = require("../shared/use-theme-classes.js");

var _useTouchRipple = require("../shared/use-touch-ripple.js");

var _useDarkClasses = require("../shared/use-dark-classes.js");

function render(_ctx, _cache) {
  return (0, _vue.openBlock)(), (0, _vue.createBlock)((0, _vue.resolveDynamicComponent)(_ctx.component), {
    ref: "rippleElRef",
    class: (0, _vue.normalizeClass)(_ctx.classes)
  }, {
    default: (0, _vue.withCtx)(() => [_ctx.theme === 'material' && _ctx.tabbar ? ((0, _vue.openBlock)(), (0, _vue.createElementBlock)("span", {
      key: 0,
      class: (0, _vue.normalizeClass)(_ctx.c.tabbarHighlight[_ctx.tabbarState])
    }, null, 2)) : (0, _vue.createCommentVNode)("", true), (0, _vue.renderSlot)(_ctx.$slots, "default")]),
    _: 3
  }, 8, ["class"]);
}

var _default = {
  name: 'twm-link',
  render,
  props: {
    component: {
      type: String,
      default: 'a'
    },
    colors: {
      type: Object
    },
    ios: {
      type: Boolean,
      default: undefined
    },
    material: {
      type: Boolean,
      default: undefined
    },
    // Toolbar/navbar link
    navbar: Boolean,
    toolbar: Boolean,
    iconOnly: Boolean,
    tabbar: Boolean,
    tabbarActive: Boolean,
    touchRipple: {
      type: Boolean,
      default: true
    }
  },

  setup(props, ctx) {
    const rippleElRef = (0, _vue.ref)(null);
    const theme = (0, _useTheme.useTheme)(props);
    const needsTouchRipple = (0, _vue.computed)(() => theme.value === 'material' && props.touchRipple && (props.toolbar || props.tabbar || props.navbar));
    (0, _useTouchRipple.useTouchRipple)(rippleElRef, props, () => props.toolbar || props.tabbar || props.navbar);
    const colors = (0, _vue.computed)(() => ({
      text: 'text-primary',
      tabbarInactive: (0, _cls.cls)(`text-black`, (0, _useDarkClasses.useDarkClasses)('dark:text-white dark:text-opacity-55')),
      ...(props.colors || {})
    }));
    const textColor = (0, _vue.computed)(() => props.tabbar && !props.tabbarActive ? colors.value.tabbarInactive : colors.value.text);
    const tabbarState = (0, _vue.computed)(() => props.tabbarActive ? 'active' : 'inactive');
    const c = (0, _useThemeClasses.useThemeClasses)(props, () => ({
      base: {
        common: (0, _cls.cls)(textColor.value, `inline-flex space-x-1 justify-center items-center cursor-pointer select-none`, needsTouchRipple.value && `touch-ripple-primary ${(0, _positionClass.positionClass)('relative', ctx.attrs.class)} z-10`),
        notTabbar: {
          ios: `active:opacity-30 duration-300 active:duration-0`,
          material: needsTouchRipple.value ? '' : `active:opacity-55`
        }
      },
      tabbar: {
        common: (0, _cls.cls)((0, _positionClass.positionClass)('relative', ctx.attrs.class), `w-full h-full duration-300`),
        material: 'uppercase font-medium text-sm overflow-hidden z-10',
        active: {},
        inactive: {
          ios: 'text-opacity-40',
          material: 'text-opacity-55'
        }
      },
      tabbarHighlight: {
        material: 'absolute left-0 top-0 w-full h-0.5 bg-primary duration-300',
        active: 'opacity-100',
        inactive: 'opacity-0'
      },
      toolbar: {
        common: (0, _cls.cls)(`h-full max-h-12`, props.iconOnly && 'touch-ripple-inset'),
        material: 'px-3'
      },
      navbar: {
        common: (0, _cls.cls)(`h-full max-h-12`, props.iconOnly && 'touch-ripple-inset'),
        material: 'px-3'
      }
    }));
    const classes = (0, _vue.computed)(() => (0, _cls.cls)( // base
    c.value.base[props.tabbar ? 'default' : 'notTabbar'], props.toolbar && c.value.toolbar, props.navbar && c.value.navbar, props.tabbar && c.value.tabbar[tabbarState.value]));
    return {
      c,
      classes,
      rippleElRef,
      theme,
      tabbarState
    };
  }

};
exports.default = _default;
