// tailwind-mobile/vue/cjs/components/Page.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _vue = require("vue");

var _cls = require("../shared/cls.js");

var _positionClass = require("../shared/position-class.js");

var _useDarkClasses = require("../shared/use-dark-classes.js");

var _useThemeClasses = require("../shared/use-theme-classes.js");

function render(_ctx, _cache) {
  return (0, _vue.openBlock)(), (0, _vue.createBlock)((0, _vue.resolveDynamicComponent)(_ctx.component), {
    class: (0, _vue.normalizeClass)(_ctx.c.base)
  }, {
    default: (0, _vue.withCtx)(() => [(0, _vue.renderSlot)(_ctx.$slots, "default")]),
    _: 3
  }, 8, ["class"]);
}

var _default = {
  name: 'twm-page',
  render,
  props: {
    component: {
      type: String,
      default: 'div'
    },
    ios: {
      type: Boolean,
      default: undefined
    },
    material: {
      type: Boolean,
      default: undefined
    }
  },

  setup(props, ctx) {
    const c = (0, _useThemeClasses.useThemeClasses)(props, () => ({
      base: {
        common: (0, _cls.cls)('h-full w-full left-0 top-0 overflow-auto', (0, _positionClass.positionClass)('absolute', ctx.attrs.class)),
        ios: (0, _cls.cls)('bg-page-ios-light', (0, _useDarkClasses.useDarkClasses)('dark:bg-page-ios-dark')),
        material: (0, _cls.cls)('bg-page-material-light', (0, _useDarkClasses.useDarkClasses)('dark:bg-page-material-dark'))
      }
    }));
    return {
      c
    };
  }

};
exports.default = _default;
