// tailwind-mobile/vue/cjs/components/Panel.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _vue = require("vue");

var _cls = require("../shared/cls.js");

var _positionClass = require("../shared/position-class.js");

var _useDarkClasses = require("../shared/use-dark-classes.js");

var _useThemeClasses = require("../shared/use-theme-classes.js");

function render(_ctx, _cache) {
  return (0, _vue.openBlock)(), (0, _vue.createElementBlock)(_vue.Fragment, null, [_ctx.backdrop ? ((0, _vue.openBlock)(), (0, _vue.createElementBlock)("div", {
    key: 0,
    class: (0, _vue.normalizeClass)(_ctx.c.backdrop[_ctx.state]),
    onClick: _cache[0] || (_cache[0] = (...args) => _ctx.onBackdropClick && _ctx.onBackdropClick(...args))
  }, null, 2)) : (0, _vue.createCommentVNode)("", true), ((0, _vue.openBlock)(), (0, _vue.createBlock)((0, _vue.resolveDynamicComponent)(_ctx.component), {
    class: (0, _vue.normalizeClass)(_ctx.classes)
  }, {
    default: (0, _vue.withCtx)(() => [(0, _vue.renderSlot)(_ctx.$slots, "default")]),
    _: 3
  }, 8, ["class"]))], 64);
}

var _default = {
  name: 'twm-panel',
  render,
  props: {
    component: {
      type: String,
      default: 'div'
    },
    colors: {
      type: Object
    },
    ios: {
      type: Boolean,
      default: undefined
    },
    material: {
      type: Boolean,
      default: undefined
    },
    size: {
      type: String,
      default: 'w-72 h-screen'
    },
    side: {
      type: String,
      default: 'left'
    },
    opened: Boolean,
    backdrop: {
      type: Boolean,
      default: true
    }
  },
  emits: ['backdropclick'],

  setup(props, ctx) {
    const state = (0, _vue.computed)(() => props.opened ? 'opened' : 'closed');
    const colors = (0, _vue.computed)(() => ({
      bg: (0, _cls.cls)('bg-white', (0, _useDarkClasses.useDarkClasses)('dark:bg-black')),
      ...(props.colors || {})
    }));
    const c = (0, _useThemeClasses.useThemeClasses)(props, () => ({
      base: {
        common: (0, _cls.cls)('top-0 transition-transform transform duration-400 z-40 max-w-full max-h-full overflow-hidden', colors.value.bg, (0, _positionClass.positionClass)('fixed', ctx.attrs.class), props.size),
        ios: '',
        material: 'shadow-2xl'
      },
      left: {
        common: (0, _cls.cls)('no-safe-areas-right left-0'),
        opened: 'translate-x-0',
        closed: '-translate-x-full'
      },
      right: {
        common: (0, _cls.cls)('no-safe-areas-left left-full'),
        opened: '-translate-x-full',
        closed: 'translate-x-0'
      },
      backdrop: {
        common: 'fixed z-40 w-full h-full left-0 top-0 bg-black bg-opacity-50 duration-400',
        opened: '',
        closed: 'opacity-0 pointer-events-none'
      }
    }));
    const classes = (0, _vue.computed)(() => (0, _cls.cls)(c.value.base, c.value[props.side][state.value], ctx.attrs.class));

    const onBackdropClick = e => {
      ctx.emit('backdropclick', e);
    };

    return {
      onBackdropClick,
      c,
      state,
      classes
    };
  }

};
exports.default = _default;
