// tailwind-mobile/vue/cjs/components/Range.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _vue = require("vue");

var _cls = require("../shared/cls.js");

var _useDarkClasses = require("../shared/use-dark-classes.js");

var _useThemeClasses = require("../shared/use-theme-classes.js");

const _hoisted_1 = ["id", "name", "min", "max", "step", "value", "readonly", "disabled"];

function render(_ctx, _cache) {
  return (0, _vue.openBlock)(), (0, _vue.createBlock)((0, _vue.resolveDynamicComponent)(_ctx.component), {
    class: (0, _vue.normalizeClass)(_ctx.c.base)
  }, {
    default: (0, _vue.withCtx)(() => [(0, _vue.createElementVNode)("span", {
      class: (0, _vue.normalizeClass)(_ctx.c.trackBg)
    }, null, 2), (0, _vue.createElementVNode)("span", {
      class: (0, _vue.normalizeClass)(_ctx.c.trackValue),
      style: (0, _vue.normalizeStyle)({
        width: `${_ctx.valueWidth}%`
      })
    }, null, 6), (0, _vue.createElementVNode)("input", {
      id: _ctx.inputId,
      class: (0, _vue.normalizeClass)(_ctx.cls(_ctx.c.input, _ctx.c.inputThumb, _ctx.c.inputTrack)),
      type: "range",
      name: _ctx.name,
      min: _ctx.min,
      max: _ctx.max,
      step: _ctx.step,
      value: _ctx.value,
      readonly: _ctx.readonly,
      disabled: _ctx.disabled,
      onInput: _cache[0] || (_cache[0] = (...args) => _ctx.onInput && _ctx.onInput(...args)),
      onChange: _cache[1] || (_cache[1] = (...args) => _ctx.onChange && _ctx.onChange(...args)),
      onFocus: _cache[2] || (_cache[2] = (...args) => _ctx.onFocus && _ctx.onFocus(...args)),
      onBlur: _cache[3] || (_cache[3] = (...args) => _ctx.onBlur && _ctx.onBlur(...args))
    }, null, 42, _hoisted_1)]),
    _: 1
  }, 8, ["class"]);
}

var _default = {
  name: 'twm-range',
  render,
  props: {
    component: {
      type: String,
      default: 'div'
    },
    colors: {
      type: Object
    },
    ios: {
      type: Boolean,
      default: undefined
    },
    material: {
      type: Boolean,
      default: undefined
    },
    inputId: {
      type: String,
      default: undefined
    },
    name: {
      type: String,
      default: undefined
    },
    value: {
      type: Number,
      default: undefined
    },
    readonly: {
      type: Boolean,
      default: false
    },
    disabled: {
      type: Boolean,
      default: false
    },
    step: {
      type: Number,
      default: 1
    },
    min: {
      type: Number,
      default: 0
    },
    max: {
      type: Number,
      default: 100
    }
  },
  emits: ['input', 'change', 'focus', 'blur'],

  setup(props, ctx) {
    const colors = (0, _vue.computed)(() => ({
      valueBg: 'bg-primary',
      thumbBgIos: 'range-thumb:bg-white',
      thumbBgMaterial: 'range-thumb:bg-primary',
      ...(props.colors || {})
    }));
    const c = (0, _useThemeClasses.useThemeClasses)(props, () => ({
      base: {
        common: 'block relative select-none w-full self-center touch-pan-y',
        ios: 'h-7',
        material: 'h-5'
      },
      trackBg: {
        common: (0, _cls.cls)('absolute top-1/2 left-0 w-full transform -translate-y-1/2 bg-black bg-opacity-20', (0, _useDarkClasses.useDarkClasses)('dark:bg-white dark:bg-opacity-20')),
        ios: 'h-1 rounded-full',
        material: 'h-0.5'
      },
      trackValue: {
        common: `absolute top-1/2 left-0 w-full transform -translate-y-1/2 ${colors.value.valueBg}`,
        ios: 'h-1 rounded-full',
        material: 'h-0.5'
      },
      input: {
        common: 'appearance-none w-full bg-transparent cursor-pointer block focus:outline-none relative',
        ios: 'h-7',
        material: 'h-5'
      },
      inputThumb: {
        common: 'range-thumb:relative range-thumb:appearance-none range-thumb:rounded-full range-thumb:border-none',
        ios: `range-thumb:w-7 range-thumb:h-7 range-thumb:-mt-3.5 ${colors.value.thumbBgIos} range-thumb:shadow-ios-toggle`,
        material: `range-thumb:w-3 range-thumb:h-3 ${colors.value.thumbBgMaterial} range-thumb:-mt-1.5 range-thumb:transform range-thumb:duration-200 active:range-thumb:scale-150`
      },
      inputTrack: {
        common: 'range-track:appearance-none range-track:h-px'
      }
    }));
    const valueWidth = (0, _vue.computed)(() => ((props.value || 0) - props.min) / (props.max - props.min) * 100);

    const onInput = e => {
      ctx.emit('input', e);
    };

    const onChange = e => {
      ctx.emit('change', e);
    };

    const onFocus = e => {
      ctx.emit('focus', e);
    };

    const onBlur = e => {
      ctx.emit('blur', e);
    };

    return {
      c,
      valueWidth,
      cls: _cls.cls,
      onInput,
      onChange,
      onFocus,
      onBlur
    };
  }

};
exports.default = _default;
