// tailwind-mobile/vue/cjs/components/Segmented.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _vue = require("vue");

var _cls = require("../shared/cls.js");

var _useDarkClasses = require("../shared/use-dark-classes.js");

var _useThemeClasses = require("../shared/use-theme-classes.js");

function render(_ctx, _cache) {
  return (0, _vue.openBlock)(), (0, _vue.createBlock)((0, _vue.resolveDynamicComponent)(_ctx.component), {
    class: (0, _vue.normalizeClass)(_ctx.classes)
  }, {
    default: (0, _vue.withCtx)(() => [_ctx.outline ? ((0, _vue.openBlock)(), (0, _vue.createElementBlock)("span", {
      key: 0,
      class: (0, _vue.normalizeClass)(_ctx.c.outlineInner)
    }, [(0, _vue.renderSlot)(_ctx.$slots, "default")], 2)) : (0, _vue.renderSlot)(_ctx.$slots, "default", {
      key: 1
    }), _ctx.strong ? ((0, _vue.openBlock)(), (0, _vue.createElementBlock)("span", {
      key: 2,
      class: (0, _vue.normalizeClass)(_ctx.c.strongHighlight),
      style: (0, _vue.normalizeStyle)(_ctx.getHighlightStyle())
    }, null, 6)) : (0, _vue.createCommentVNode)("", true)]),
    _: 3
  }, 8, ["class"]);
}

var _default = {
  name: 'twm-segmented',
  render,
  props: {
    component: {
      type: String,
      default: 'div'
    },
    colors: {
      type: Object
    },
    ios: {
      type: Boolean,
      default: undefined
    },
    material: {
      type: Boolean,
      default: undefined
    },
    // Style props
    raised: {
      type: Boolean,
      default: false
    },
    outline: {
      type: Boolean,
      default: false
    },
    strong: {
      type: Boolean,
      default: false
    },
    rounded: {
      type: Boolean,
      default: false
    },
    activeButtonIndex: {
      type: Number,
      default: undefined
    },
    childButtonsLength: {
      type: Number,
      default: undefined
    }
  },

  setup(props, ctx) {
    const colors = (0, _vue.computed)(() => ({
      border: 'border-primary',
      divide: 'divide-primary',
      ...(props.colors || {})
    }));
    const c = (0, _useThemeClasses.useThemeClasses)(props, () => ({
      base: {
        common: `flex justify-center items-center overflow-hidden w-full`,
        square: 'rounded',
        rounded: 'rounded-full'
      },
      raised: props.outline ? 'shadow' : `shadow divide-x divide-black divide-opacity-10`,
      outline: `border-2 ${colors.value.border}`,
      outlineInner: `-m-0.5 flex w-full justify-center items-center divide-x-2 ${colors.value.divide}`,
      strong: (0, _cls.cls)('p-0.5 bg-black bg-opacity-5 space-x-1 relative', (0, _useDarkClasses.useDarkClasses)('dark:bg-white dark:bg-opacity-10')),
      strongHighlight: (0, _cls.cls)('bg-red-500 absolute left-0.5 top-0.5 bottom-0.5 !ml-0 pointer-events-none duration-200', props.rounded ? 'rounded-full' : 'rounded', `bg-white shadow`, (0, _useDarkClasses.useDarkClasses)('dark:bg-opacity-15 dark:touch-ripple-white'))
    }));
    const classes = (0, _vue.computed)(() => (0, _cls.cls)( // base
    props.rounded ? c.value.base.rounded : c.value.base.square, props.raised && c.value.raised, props.outline && c.value.outline, props.strong && c.value.strong));

    const getHighlightStyle = () => {
      let buttonsLength = props.childButtonsLength;
      let activeIndex = props.activeButtonIndex;
      const children = ctx.slots && ctx.slots.default && ctx.slots.default();

      if (typeof activeIndex === 'undefined' && children && children.length) {
        if (typeof buttonsLength === 'undefined') {
          buttonsLength = children.length || 0;
        }

        const activeButton = children.filter(child => child.props && (child.props.active || child.props.active === '' || child.props.segmentedActive || child.props.segmentedActive === ''))[0];
        activeIndex = children.indexOf(activeButton);
      }

      const between = '4px';
      const padding = '2px';
      const highlightWidth = `calc((100% - ${padding} * 2 - ${between} * (${buttonsLength} - 1)) / ${buttonsLength})`;
      const highlightTranslate = `calc(${activeIndex} * 100% + ${activeIndex} * ${between})`;
      return {
        width: highlightWidth,
        transform: `translateX(${highlightTranslate})`
      };
    };

    return {
      c,
      classes,
      getHighlightStyle
    };
  }

};
exports.default = _default;
