// tailwind-mobile/vue/cjs/components/Stepper.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _vue = require("vue");

var _cls = require("../shared/cls.js");

var _useThemeClasses = require("../shared/use-theme-classes.js");

var _useTouchRipple = require("../shared/use-touch-ripple.js");

const _hoisted_1 = ["placeholder", "type", "value", "disabled", "readonly"];

function render(_ctx, _cache) {
  return (0, _vue.openBlock)(), (0, _vue.createBlock)((0, _vue.resolveDynamicComponent)(_ctx.component), {
    class: (0, _vue.normalizeClass)(_ctx.classes)
  }, {
    default: (0, _vue.withCtx)(() => [(0, _vue.createElementVNode)("span", {
      ref: "buttonLeftElRef",
      class: (0, _vue.normalizeClass)(_ctx.buttonLeftClasses),
      onClick: _cache[0] || (_cache[0] = (...args) => _ctx.onMinus && _ctx.onMinus(...args))
    }, [(0, _vue.createElementVNode)("span", {
      class: (0, _vue.normalizeClass)(_ctx.c.hBar)
    }, null, 2)], 2), _ctx.input && !_ctx.buttonsOnly ? ((0, _vue.openBlock)(), (0, _vue.createElementBlock)("input", {
      key: 0,
      class: (0, _vue.normalizeClass)(_ctx.valueClasses),
      placeholder: _ctx.inputPlaceholder,
      type: _ctx.inputType,
      value: _ctx.value,
      disabled: _ctx.inputDisabled,
      readonly: _ctx.inputReadonly,
      onInput: _cache[1] || (_cache[1] = (...args) => _ctx.onInput && _ctx.onInput(...args)),
      onChange: _cache[2] || (_cache[2] = (...args) => _ctx.onChange && _ctx.onChange(...args)),
      onFocus: _cache[3] || (_cache[3] = (...args) => _ctx.onFocus && _ctx.onFocus(...args)),
      onBlur: _cache[4] || (_cache[4] = (...args) => _ctx.onBlur && _ctx.onBlur(...args))
    }, null, 42, _hoisted_1)) : (0, _vue.createCommentVNode)("", true), !_ctx.input && !_ctx.buttonsOnly ? ((0, _vue.openBlock)(), (0, _vue.createElementBlock)("span", {
      key: 1,
      class: (0, _vue.normalizeClass)(_ctx.valueClasses)
    }, (0, _vue.toDisplayString)(_ctx.value), 3)) : (0, _vue.createCommentVNode)("", true), (0, _vue.createElementVNode)("span", {
      ref: "buttonRightElRef",
      class: (0, _vue.normalizeClass)(_ctx.buttonRightClasses),
      onClick: _cache[5] || (_cache[5] = (...args) => _ctx.onPlus && _ctx.onPlus(...args))
    }, [(0, _vue.createElementVNode)("span", {
      class: (0, _vue.normalizeClass)(_ctx.c.hBar)
    }, null, 2), (0, _vue.createElementVNode)("span", {
      class: (0, _vue.normalizeClass)(_ctx.c.vBar)
    }, null, 2)], 2)]),
    _: 1
  }, 8, ["class"]);
}

var _default = {
  name: 'twm-stepper',
  render,
  props: {
    component: {
      type: String,
      default: 'span'
    },
    colors: {
      type: Object
    },
    ios: {
      type: Boolean,
      default: undefined
    },
    material: {
      type: Boolean,
      default: undefined
    },
    value: {
      type: Number,
      default: 0
    },
    input: {
      type: Boolean,
      default: false
    },
    inputType: {
      type: String,
      default: 'text'
    },
    inputPlaceholder: String,
    inputDisabled: Boolean,
    inputReadonly: Boolean,
    buttonsOnly: Boolean,
    rounded: Boolean,
    small: Boolean,
    large: Boolean,
    raised: Boolean,
    outline: Boolean,
    touchRipple: {
      type: Boolean,
      default: true
    }
  },
  emits: ['input', 'change', 'focus', 'blur', 'minus', 'plus'],

  setup(props, ctx) {
    const buttonLeftElRef = (0, _vue.ref)(null);
    const buttonRightElRef = (0, _vue.ref)(null);
    (0, _useTouchRipple.useTouchRipple)(buttonLeftElRef, props);
    (0, _useTouchRipple.useTouchRipple)(buttonRightElRef, props);
    const colors = (0, _vue.computed)(() => ({
      text: 'text-primary',
      border: 'border-primary',
      bg: 'bg-primary',
      activeBg: 'active:bg-primary',
      activeBgDark: 'active:bg-primary-dark',
      touchRipple: 'touch-ripple-primary',
      ...(props.colors || {})
    }));
    const size = (0, _vue.computed)(() => props.large ? 'large' : props.small ? 'small' : 'medium');
    const style = (0, _vue.computed)(() => props.outline && props.raised ? 'clear' : props.outline ? 'outline' : 'fill');
    const shape = (0, _vue.computed)(() => props.rounded ? 'rounded' : 'square');
    const c = (0, _useThemeClasses.useThemeClasses)(props, () => ({
      base: `${colors.value.text} inline-flex items-stretch`,
      raised: 'shadow',
      size: {
        small: 'h-7',
        medium: {
          ios: `h-7`,
          material: `h-9`
        },
        large: {
          ios: `h-11`,
          material: `h-12`
        }
      },
      shape: {
        square: 'rounded',
        rounded: 'rounded-full'
      },
      button: {
        common: 'relative flex items-center justify-center w-10 cursor-pointer overflow-hidden z-10 select-none'
      },
      buttonLeftShape: {
        square: 'rounded-l',
        rounded: 'rounded-l-full'
      },
      buttonRightShape: {
        square: 'rounded-r',
        rounded: 'rounded-r-full'
      },
      buttonStyle: {
        fill: (0, _cls.cls)('text-white', colors.value.bg, colors.value.activeBgDark, 'touch-ripple-white', props.buttonsOnly && 'first:border-r border-black border-opacity-10'),
        outline: (0, _cls.cls)('border-2', colors.value.border, colors.value.text, colors.value.touchRipple, colors.value.activeBg, 'active:bg-opacity-15', props.buttonsOnly && 'first:border-r-0'),
        clear: `${colors.value.text} ${colors.value.activeBg} ${colors.value.touchRipple} active:bg-opacity-15 last:border-l border-black border-opacity-10`
      },
      input: {
        common: 'focus:outline-none text-center appearance-none bg-transparent'
      },
      value: {
        common: `w-11 flex items-center justify-center ${colors.value.text} font-medium`,
        ios: 'text-base',
        material: 'text-sm',
        fill: `border-t-2 border-b-2 ${colors.value.border} ${colors.value.text}`,
        outline: `border-t-2 border-b-2 ${colors.value.border} ${colors.value.text}`,
        clear: 'border-l border-black border-opacity-10'
      },
      hBar: 'w-4 h-0.5 bg-current block',
      vBar: 'w-0.5 h-4 bg-current block absolute inset-1/2 transform -translate-x-1/2 -translate-y-1/2'
    }));
    const classes = (0, _vue.computed)(() => (0, _cls.cls)(c.value.base, props.raised && c.value.raised, c.value.size[size.value], c.value.shape[shape.value]));
    const buttonLeftClasses = (0, _vue.computed)(() => (0, _cls.cls)(c.value.button, c.value.buttonStyle[style.value], c.value.buttonLeftShape[shape.value]));
    const buttonRightClasses = (0, _vue.computed)(() => (0, _cls.cls)(c.value.button, c.value.buttonStyle[style.value], c.value.buttonRightShape[shape.value]));
    const valueClasses = (0, _vue.computed)(() => (0, _cls.cls)(props.input && c.value.input, c.value.value[style.value]));

    const onMinus = e => {
      ctx.emit('minus', e);
    };

    const onInput = e => {
      ctx.emit('input', e);
    };

    const onChange = e => {
      ctx.emit('change', e);
    };

    const onFocus = e => {
      ctx.emit('focus', e);
    };

    const onBlur = e => {
      ctx.emit('blur', e);
    };

    const onPlus = e => {
      ctx.emit('plus', e);
    };

    return {
      c,
      buttonLeftElRef,
      buttonRightElRef,
      classes,
      buttonLeftClasses,
      buttonRightClasses,
      valueClasses,
      onMinus,
      onInput,
      onChange,
      onFocus,
      onBlur,
      onPlus
    };
  }

};
exports.default = _default;
