// tailwind-mobile/vue/cjs/components/Toast.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _vue = require("vue");

var _cls = require("../shared/cls.js");

var _positionClass = require("../shared/position-class.js");

var _useThemeClasses = require("../shared/use-theme-classes.js");

function render(_ctx, _cache) {
  return (0, _vue.openBlock)(), (0, _vue.createBlock)((0, _vue.resolveDynamicComponent)(_ctx.component), {
    class: (0, _vue.normalizeClass)(_ctx.c.base[_ctx.position])
  }, {
    default: (0, _vue.withCtx)(() => [(0, _vue.createElementVNode)("div", {
      class: (0, _vue.normalizeClass)(_ctx.c.content)
    }, [(0, _vue.renderSlot)(_ctx.$slots, "default"), _ctx.slots.button ? ((0, _vue.openBlock)(), (0, _vue.createElementBlock)("div", {
      key: 0,
      class: (0, _vue.normalizeClass)(_ctx.c.button)
    }, [(0, _vue.renderSlot)(_ctx.$slots, "button")], 2)) : (0, _vue.createCommentVNode)("", true)], 2)]),
    _: 3
  }, 8, ["class"]);
}

var _default = {
  name: 'twm-toast',
  render,
  props: {
    component: {
      type: String,
      default: 'div'
    },
    colors: {
      type: Object
    },
    ios: {
      type: Boolean,
      default: undefined
    },
    material: {
      type: Boolean,
      default: undefined
    },
    position: {
      type: String,
      default: 'left'
    },
    opened: Boolean,
    translucent: {
      type: Boolean,
      default: true
    }
  },

  setup(props, ctx) {
    const colors = (0, _vue.computed)(() => ({
      bgIos: 'bg-toast-ios',
      bgMaterial: 'bg-toast-material',
      text: 'text-white',
      ...(props.colors || {})
    }));
    const c = (0, _useThemeClasses.useThemeClasses)(props, () => ({
      base: {
        common: (0, _cls.cls)(colors.value.text, (0, _positionClass.positionClass)('fixed', ctx.attrs.class), 'z-50 transform transition-transform-opacity w-full sm:w-auto left-0 bottom-0 sm:max-w-lg', !props.opened && 'translate-y-full opacity-0 pointer-events-none'),
        ios: (0, _cls.cls)(`sm:rounded-lg duration-300`, colors.value.bgIos, props.translucent && 'translucent'),
        material: (0, _cls.cls)(`sm:rounded duration-200`, colors.value.bgMaterial),
        left: 'sm:left-4-safe sm:bottom-4-safe',
        right: 'sm:right-4-safe sm:bottom-4-safe sm:left-auto',
        center: 'sm:left-1/2 sm:bottom-4-safe sm:-translate-x-1/2'
      },
      content: {
        common: 'flex items-center justify-between',
        ios: 'pl-4-safe pr-4-safe pt-3 pb-3-safe sm:px-4 sm:py-3',
        material: 'px-6 py-3.5'
      },
      button: {
        common: '-my-2 flex-shrink-0 ml-4',
        ios: '',
        material: '-mr-2'
      }
    }));
    return {
      c,
      slots: ctx.slots
    };
  }

};
exports.default = _default;
