// tailwind-mobile/vue/cjs/components/Toggle.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _vue = require("vue");

var _cls = require("../shared/cls.js");

var _positionClass = require("../shared/position-class.js");

var _useDarkClasses = require("../shared/use-dark-classes.js");

var _useThemeClasses = require("../shared/use-theme-classes.js");

const _hoisted_1 = ["name", "value", "disabled", "readonly", "checked"];

function render(_ctx, _cache) {
  return (0, _vue.openBlock)(), (0, _vue.createBlock)((0, _vue.resolveDynamicComponent)(_ctx.component), {
    class: (0, _vue.normalizeClass)(_ctx.c.base[_ctx.state])
  }, {
    default: (0, _vue.withCtx)(() => [(0, _vue.createElementVNode)("input", {
      type: "checkbox",
      name: _ctx.name,
      value: _ctx.value,
      disabled: _ctx.disabled,
      readonly: _ctx.readonly,
      checked: _ctx.checked,
      class: (0, _vue.normalizeClass)(_ctx.c.input)
    }, null, 10, _hoisted_1), (0, _vue.createElementVNode)("span", {
      class: (0, _vue.normalizeClass)(_ctx.c.inner[_ctx.state])
    }, null, 2), (0, _vue.createElementVNode)("span", {
      class: (0, _vue.normalizeClass)(_ctx.c.knob[_ctx.state])
    }, null, 2), (0, _vue.renderSlot)(_ctx.$slots, "default")]),
    _: 3
  }, 8, ["class"]);
}

var _default = {
  name: 'twm-toggle',
  render,
  props: {
    component: {
      type: String,
      default: 'label'
    },
    colors: {
      type: Object
    },
    ios: {
      type: Boolean,
      default: undefined
    },
    material: {
      type: Boolean,
      default: undefined
    },
    checked: {
      type: Boolean,
      default: false
    },
    name: {
      type: String,
      default: undefined
    },
    value: {
      type: [Number, String],
      default: undefined
    },
    disabled: {
      type: Boolean,
      default: false
    },
    readonly: {
      type: Boolean,
      default: false
    }
  },

  setup(props, ctx) {
    const colors = (0, _vue.computed)(() => ({
      bg: 'bg-primary',
      ...(props.colors || {})
    }));
    const state = (0, _vue.computed)(() => props.checked ? 'checked' : 'notChecked');
    const c = (0, _useThemeClasses.useThemeClasses)(props, () => ({
      base: {
        common: (0, _cls.cls)((0, _positionClass.positionClass)('relative', ctx.attrs.class), `cursor-pointer select-none inline-block align-middle relative duration-300 rounded-full`),
        ios: 'h-8 w-13 p-0.5',
        material: 'w-9 h-3.5',
        notChecked: {
          ios: (0, _cls.cls)('bg-black bg-opacity-10', (0, _useDarkClasses.useDarkClasses)('dark:bg-white dark:bg-opacity-20')),
          material: (0, _cls.cls)('bg-black bg-opacity-30', (0, _useDarkClasses.useDarkClasses)('dark:bg-white dark:bg-opacity-30'))
        },
        checked: {
          ios: `${colors.value.bg}`,
          material: `${colors.value.bg} bg-opacity-50`
        }
      },
      inner: {
        ios: (0, _cls.cls)('w-full h-full bg-white rounded-full block duration-300 transform', (0, _useDarkClasses.useDarkClasses)('dark:bg-opacity-0')),
        notChecked: {},
        checked: {
          ios: 'scale-0'
        }
      },
      knob: {
        common: 'rounded-full absolute duration-300 transform',
        ios: 'w-7 h-7 shadow-ios-toggle left-0.5 top-0.5 bg-white',
        material: 'w-5 h-5 shadow-md left-0 -top-0.75',
        checked: {
          ios: 'translate-x-5',
          material: `${colors.value.bg} translate-x-4`
        },
        notChecked: {
          material: 'bg-white'
        }
      },
      input: {
        common: 'hidden'
      }
    }));
    return {
      c,
      state
    };
  }

};
exports.default = _default;
