// tailwind-mobile/vue/cjs/components/Toolbar.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _vue = require("vue");

var _cls = require("../shared/cls.js");

var _positionClass = require("../shared/position-class.js");

var _useThemeClasses = require("../shared/use-theme-classes.js");

var _useDarkClasses = require("../shared/use-dark-classes.js");

function render(_ctx, _cache) {
  return (0, _vue.openBlock)(), (0, _vue.createBlock)((0, _vue.resolveDynamicComponent)(_ctx.component), {
    class: (0, _vue.normalizeClass)(_ctx.c.base)
  }, {
    default: (0, _vue.withCtx)(() => [(0, _vue.createElementVNode)("div", {
      class: (0, _vue.normalizeClass)(_ctx.c.bg)
    }, null, 2), (0, _vue.createElementVNode)("div", {
      class: (0, _vue.normalizeClass)(_ctx.c.inner[_ctx.tabbar && _ctx.tabbarLabels ? 'tabbarLabels' : 'toolbar'])
    }, [(0, _vue.renderSlot)(_ctx.$slots, "default")], 2)]),
    _: 3
  }, 8, ["class"]);
}

var _default = {
  name: 'twm-toolbar',
  render,
  props: {
    component: {
      type: String,
      default: 'div'
    },
    colors: {
      type: Object
    },
    ios: {
      type: Boolean,
      default: undefined
    },
    material: {
      type: Boolean,
      default: undefined
    },
    translucent: {
      type: Boolean,
      default: true
    },
    bgClass: {
      type: String,
      default: ''
    },
    innerClass: {
      type: String,
      default: ''
    },
    hairlines: {
      type: Boolean,
      default: true
    },
    tabbar: {
      type: Boolean,
      default: false
    },
    tabbarLabels: {
      type: Boolean,
      default: false
    },
    top: {
      type: Boolean,
      default: false
    }
  },

  setup(props, ctx) {
    const colors = (0, _vue.computed)(() => ({
      bgIos: (0, _cls.cls)('bg-bars-ios-light', (0, _useDarkClasses.useDarkClasses)('dark:bg-bars-ios-dark')),
      bgMaterial: (0, _cls.cls)('bg-bars-material-light', (0, _useDarkClasses.useDarkClasses)('dark:bg-bars-material-dark')),
      ...(props.colors || {})
    }));
    const c = (0, _useThemeClasses.useThemeClasses)(props, () => ({
      base: {
        common: (0, _cls.cls)(`w-full z-20`, (0, _positionClass.positionClass)('relative', ctx.attrs.class), !props.top && 'pb-safe')
      },
      bg: {
        common: (0, _cls.cls)('absolute w-full h-full left-0 top-0', props.bgClass),
        ios: (0, _cls.cls)(props.hairlines && (props.top ? 'hairline-b' : 'hairline-t'), colors.value.bgIos, props.translucent && 'translucent'),
        material: (0, _cls.cls)(`shadow-md ${colors.value.bgMaterial}`, !props.top && 'transform rotate-180')
      },
      inner: {
        common: (0, _cls.cls)(`flex relative justify-between items-center w-full overflow-hidden`, props.innerClass),
        ios: 'pl-2-safe pr-2-safe',
        material: !props.tabbar ? 'pl-2-safe pr-2-safe' : '',
        toolbar: {
          ios: 'h-11',
          material: 'h-12'
        },
        tabbarLabels: {
          ios: 'h-12.5',
          material: 'h-14'
        }
      }
    }));
    return {
      c
    };
  }

};
exports.default = _default;
