// tailwind-mobile/vue/esm/components/BlockTitle.js
import { renderSlot as _renderSlot, resolveDynamicComponent as _resolveDynamicComponent, normalizeClass as _normalizeClass, withCtx as _withCtx, openBlock as _openBlock, createBlock as _createBlock } from "vue";

function render(_ctx, _cache) {
  return _openBlock(), _createBlock(_resolveDynamicComponent(_ctx.component), {
    class: _normalizeClass(_ctx.c.base)
  }, {
    default: _withCtx(() => [_renderSlot(_ctx.$slots, "default")]),
    _: 3
  }, 8, ["class"]);
}

import { useThemeClasses } from '../shared/use-theme-classes.js';
export default {
  name: 'twm-block-title',
  render,
  props: {
    component: {
      type: String,
      default: 'span'
    },
    ios: {
      type: Boolean,
      default: undefined
    },
    material: {
      type: Boolean,
      default: undefined
    },
    withBlock: {
      type: Boolean,
      default: true
    }
  },

  setup(props) {
    const c = useThemeClasses(props, () => ({
      base: {
        common: `pl-4-safe pr-4-safe mt-8 flex justify-between items-center ${props.withBlock ? '-mb-6' : 'mb-2'}`,
        ios: `font-semibold`,
        material: `font-medium`
      }
    }));
    return {
      c
    };
  }

};
