// tailwind-mobile/vue/esm/components/Navbar.js
import { normalizeClass as _normalizeClass, createElementVNode as _createElementVNode, renderSlot as _renderSlot, openBlock as _openBlock, createElementBlock as _createElementBlock, createCommentVNode as _createCommentVNode, toDisplayString as _toDisplayString, createTextVNode as _createTextVNode, resolveDynamicComponent as _resolveDynamicComponent, withCtx as _withCtx, createBlock as _createBlock } from "vue";

function render(_ctx, _cache) {
  return _openBlock(), _createBlock(_resolveDynamicComponent(_ctx.component), {
    class: _normalizeClass(_ctx.c.base)
  }, {
    default: _withCtx(() => [_createElementVNode("div", {
      class: _normalizeClass(_ctx.c.bg)
    }, null, 2), _createElementVNode("div", {
      class: _normalizeClass(_ctx.c.inner)
    }, [_ctx.slots.left ? (_openBlock(), _createElementBlock("div", {
      key: 0,
      class: _normalizeClass(_ctx.c.left)
    }, [_renderSlot(_ctx.$slots, "left")], 2)) : _createCommentVNode("", true), _ctx.title || _ctx.subtitle || _ctx.slots.title || _ctx.slots.subtitle ? (_openBlock(), _createElementBlock("div", {
      key: 1,
      class: _normalizeClass(_ctx.c.title)
    }, [_createTextVNode(_toDisplayString(_ctx.title), 1), _renderSlot(_ctx.$slots, "title"), _ctx.subtitle || _ctx.slots.subtitle ? (_openBlock(), _createElementBlock("div", {
      key: 0,
      class: _normalizeClass(_ctx.c.subtitle)
    }, [_createTextVNode(_toDisplayString(_ctx.subtitle), 1), _renderSlot(_ctx.$slots, "subtitle")], 2)) : _createCommentVNode("", true)], 2)) : _createCommentVNode("", true), _ctx.slots.right ? (_openBlock(), _createElementBlock("div", {
      key: 2,
      class: _normalizeClass(_ctx.c.right)
    }, [_renderSlot(_ctx.$slots, "right")], 2)) : _createCommentVNode("", true), _renderSlot(_ctx.$slots, "default")], 2), _ctx.slots.subnavbar ? (_openBlock(), _createElementBlock("div", {
      key: 0,
      class: _normalizeClass(_ctx.c.subnavbar)
    }, [_renderSlot(_ctx.$slots, "subnavbar")], 2)) : _createCommentVNode("", true)]),
    _: 3
  }, 8, ["class"]);
}

import { computed } from 'vue';
import { cls } from '../shared/cls.js';
import { positionClass } from '../shared/position-class.js';
import { useThemeClasses } from '../shared/use-theme-classes.js';
import { useDarkClasses } from '../shared/use-dark-classes.js';
export default {
  name: 'twm-navbar',
  render,
  props: {
    component: {
      type: String,
      default: 'div'
    },
    colors: {
      type: Object
    },
    ios: {
      type: Boolean,
      default: undefined
    },
    material: {
      type: Boolean,
      default: undefined
    },
    bgClass: {
      type: String,
      default: ''
    },
    innerClass: {
      type: String,
      default: ''
    },
    leftClass: {
      type: String,
      default: ''
    },
    titleClass: {
      type: String,
      default: ''
    },
    subtitleClass: {
      type: String,
      default: ''
    },
    rightClass: {
      type: String,
      default: ''
    },
    subnavbarClass: {
      type: String,
      default: ''
    },
    translucent: {
      type: Boolean,
      default: true
    },
    hairlines: {
      type: Boolean,
      default: true
    },
    fontSizeIos: {
      type: String,
      default: 'text-navbar-ios'
    },
    fontSizeMaterial: {
      type: String,
      default: 'text-navbar-material'
    },
    title: String,
    subtitle: String
  },

  setup(props, ctx) {
    const colors = computed(() => ({
      bgIos: cls('bg-bars-ios-light', useDarkClasses('dark:bg-bars-ios-dark')),
      bgMaterial: cls('bg-bars-material-light', useDarkClasses('dark:bg-bars-material-dark')),
      title: cls('text-black', useDarkClasses('dark:text-white')),
      ...(props.colors || {})
    }));
    const c = useThemeClasses(props, () => ({
      base: {
        common: cls(`w-full z-20 top-0 pt-safe`, positionClass('sticky', ctx.attrs.class)),
        ios: props.fontSizeIos,
        material: props.fontSizeMaterial
      },
      bg: {
        common: cls('absolute w-full h-full left-0 top-0', props.bgClass),
        ios: cls(colors.value.bgIos, props.hairlines && 'hairline-b', props.translucent && 'translucent'),
        material: `shadow-md ${colors.value.bgMaterial}`
      },
      subnavbar: {
        common: cls('relative flex items-center', props.subnavbarClass),
        ios: 'h-11 pl-2-safe pr-2-safe',
        material: 'h-12 pl-4-safe pr-4-safe'
      },
      inner: {
        common: cls('flex relative items-center w-full overflow-hidden', props.innerClass),
        ios: cls('pl-2-safe pr-2-safe h-11', !ctx.slots.left && ctx.slots.right ? 'justify-end' : 'justify-between'),
        material: 'justify-start h-14 lg:h-16 pl-safe pr-safe'
      },
      left: {
        common: cls('flex justify-center items-center h-full', props.leftClass),
        ios: 'mr-2 transform transform-gpu',
        material: 'mx-1'
      },
      title: {
        common: cls(`${colors.value.title} whitespace-nowrap leading-tight`, props.titleClass),
        ios: 'font-semibold absolute top-1/2 left-1/2 transform-gpu -translate-x-1/2 -translate-y-1/2 text-center',
        material: 'font-medium mx-4 text-left'
      },
      subtitle: {
        common: cls('font-normal leading-none', props.subtitleClass),
        ios: 'text-2xs opacity-55',
        material: 'text-sm opacity-85'
      },
      right: {
        common: cls('flex justify-center items-center h-full', props.rightClass),
        ios: 'ml-2 transform transform-gpu',
        material: 'ml-auto mr-1'
      }
    }));
    return {
      c,
      slots: ctx.slots
    };
  }

};
