// tailwind-mobile/vue/esm/components/Toolbar.js
import { normalizeClass as _normalizeClass, createElementVNode as _createElementVNode, renderSlot as _renderSlot, resolveDynamicComponent as _resolveDynamicComponent, withCtx as _withCtx, openBlock as _openBlock, createBlock as _createBlock } from "vue";

function render(_ctx, _cache) {
  return _openBlock(), _createBlock(_resolveDynamicComponent(_ctx.component), {
    class: _normalizeClass(_ctx.c.base)
  }, {
    default: _withCtx(() => [_createElementVNode("div", {
      class: _normalizeClass(_ctx.c.bg)
    }, null, 2), _createElementVNode("div", {
      class: _normalizeClass(_ctx.c.inner[_ctx.tabbar && _ctx.tabbarLabels ? 'tabbarLabels' : 'toolbar'])
    }, [_renderSlot(_ctx.$slots, "default")], 2)]),
    _: 3
  }, 8, ["class"]);
}

import { computed } from 'vue';
import { cls } from '../shared/cls.js';
import { positionClass } from '../shared/position-class.js';
import { useThemeClasses } from '../shared/use-theme-classes.js';
import { useDarkClasses } from '../shared/use-dark-classes.js';
export default {
  name: 'twm-toolbar',
  render,
  props: {
    component: {
      type: String,
      default: 'div'
    },
    colors: {
      type: Object
    },
    ios: {
      type: Boolean,
      default: undefined
    },
    material: {
      type: Boolean,
      default: undefined
    },
    translucent: {
      type: Boolean,
      default: true
    },
    bgClass: {
      type: String,
      default: ''
    },
    innerClass: {
      type: String,
      default: ''
    },
    hairlines: {
      type: Boolean,
      default: true
    },
    tabbar: {
      type: Boolean,
      default: false
    },
    tabbarLabels: {
      type: Boolean,
      default: false
    },
    top: {
      type: Boolean,
      default: false
    }
  },

  setup(props, ctx) {
    const colors = computed(() => ({
      bgIos: cls('bg-bars-ios-light', useDarkClasses('dark:bg-bars-ios-dark')),
      bgMaterial: cls('bg-bars-material-light', useDarkClasses('dark:bg-bars-material-dark')),
      ...(props.colors || {})
    }));
    const c = useThemeClasses(props, () => ({
      base: {
        common: cls(`w-full z-20`, positionClass('relative', ctx.attrs.class), !props.top && 'pb-safe')
      },
      bg: {
        common: cls('absolute w-full h-full left-0 top-0', props.bgClass),
        ios: cls(props.hairlines && (props.top ? 'hairline-b' : 'hairline-t'), colors.value.bgIos, props.translucent && 'translucent'),
        material: cls(`shadow-md ${colors.value.bgMaterial}`, !props.top && 'transform rotate-180')
      },
      inner: {
        common: cls(`flex relative justify-between items-center w-full overflow-hidden`, props.innerClass),
        ios: 'pl-2-safe pr-2-safe',
        material: !props.tabbar ? 'pl-2-safe pr-2-safe' : '',
        toolbar: {
          ios: 'h-11',
          material: 'h-12'
        },
        tabbarLabels: {
          ios: 'h-12.5',
          material: 'h-14'
        }
      }
    }));
    return {
      c
    };
  }

};
